drivers/storage/portworx/util/util.go

```diff
@@ -304,3 +333,155 @@ func GetValueFromEnv(ctx context.Context, client client.Client, envVar *v1.EnvVa
+	}
+
+	pxService := &v1.Service{}
+	err := k8sClient.Get(
+		context.TODO(),
+		types.NamespacedName{
+			Name:      PortworxServiceName,
+			Namespace: namespace,
+		},
+		pxService,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get k8s service spec: %v", err)
+	} else if len(pxService.Spec.ClusterIP) == 0 {
+		return nil, fmt.Errorf("failed to get endpoint for portworx volume driver")
+	}
+
+	endpoint := pxService.Spec.ClusterIP
+	sdkPort := defaultSDKPort
+
+	// Get the ports from service
+	for _, pxServicePort := range pxService.Spec.Ports {
+		if pxServicePort.Name == PortworxSDKPortName && pxServicePort.Port != 0 {
+			sdkPort = int(pxServicePort.Port)
+		}
+	}
+
+	endpoint = fmt.Sprintf("%s:%d", endpoint, sdkPort)
+	return GetGrpcConn(endpoint)
+}
+
+// GetGrpcConn creates a new gRPC connection to a given endpoint
+func GetGrpcConn(endpoint string) (*grpc.ClientConn, error) {
+	dialOptions, err := GetDialOptions(IsTLSEnabled())
+	if err != nil {
+		return nil, err
+	}
+	sdkConn, err := grpcserver.Connect(endpoint, dialOptions)
+	if err != nil {
+		return nil, fmt.Errorf("error connecting to GRPC server [%s]: %v", endpoint, err)
+	}
+	return sdkConn, nil
+}
+
+// GetDialOptions is a gRPC utility to get dial options for a connection
+func GetDialOptions(tls bool) ([]grpc.DialOption, error) {
+	if !tls {
+		return []grpc.DialOption{grpc.WithInsecure()}, nil
+	}
+	capool, err := x509.SystemCertPool()
+	if err != nil {
+		return nil, fmt.Errorf("failed to load CA system certs: %v", err)
+	}
+	return []grpc.DialOption{grpc.WithTransportCredentials(
+		credentials.NewClientTLSFromCert(capool, ""),
+	)}, nil
+}
+
+// IsTLSEnabled checks if TLS is enabled for the operator
+func IsTLSEnabled() bool {
+	enabled, err := strconv.ParseBool(os.Getenv(EnvKeyPortworxEnableTLS))
+	return err == nil && enabled
+}
+
+// GetOperatorToken generates an auth token given a secret key
+func GetOperatorToken(
+	cluster *corev1alpha1.StorageCluster,
+	secretkey string,
+) (string, error) {
+	claims := &auth.Claims{
+		Issuer:  *cluster.Spec.Security.Auth.Authenticators.SelfSigned.Issuer,
+		Subject: "operator@portworx.io",
+		Name:    "operator communications",
+		Email:   "operator@portworx.io",
+		Roles:   []string{"system.admin"},
+		Groups:  []string{"*"},
+	}
+
+	signature, err := auth.NewSignatureSharedSecret(string(secretkey))
+	if err != nil {
+		return "", err
+	}
+	token, err := auth.Token(claims, signature, &auth.Options{
+		Expiration: time.Now().
+			Add(cluster.Spec.Security.Auth.Authenticators.SelfSigned.TokenLifetime.Duration).Unix(),
+	})
+	if err != nil {
+		return "", err
+	}
+
+	return token, nil
+}
+
+// GetAdminSecret gets the admin secret from a pre-configured environment variable
+func GetAdminSecret(
+	ctx context.Context,
+	cluster *corev1alpha1.StorageCluster,
+	k8sClient client.Client,
+) (string, error) {
+	var authSecret string
+	var err error
+
+	// check for provided secret
+	for _, envVar := range cluster.Spec.Env {
```

Comment:
Even here we should get the shared secret from the k8s secret that operator created (either generated or from user provided env value).